src/Core/IThreadHandling.cs

```diff
@@ -62,6 +63,24 @@ public interface IThreadHandling
         /// Switches to the background thread
         /// </summary>
         /// <remarks>Wrapper that calls <see cref="TaskScheduler.Default"/></remarks>
-        Task SwitchToBackgroundThread();
+        IAwaitableWrapper SwitchToBackgroundThread();
```

Comment:
The previous version didn't work - we need to return an awaiter, not a task.

However, we don't want to expose VS directly types in `Core`, so we wrap them.